extension/src/popup/views/ViewPublicKey/index.tsx

```diff
@@ -0,0 +1,141 @@
+
+import { Button, TextLink, Icon, Input } from "@stellar/design-system";
+
+import "./styles.scss";
+
+export const ViewPublicKey = () => {
+  const publicKey = useSelector(publicKeySelector);
+  const accountName = useSelector(accountNameSelector);
+  const [isCopied, setIsCopied] = useState(false);
+  const [isEditingName, setIsEditingName] = useState(false);
+  const { network } = useSelector(settingsNetworkDetailsSelector);
+
+  const dispatch = useDispatch();
+
+  interface FormValue {
+    accountName: string;
+  }
+
+  const initialValues: FormValue = {
+    accountName,
+  };
+
+  const handleSubmit = async (values: FormValue) => {
+    const { accountName: newAccountName } = values;
+    if (accountName !== newAccountName) {
+      await dispatch(updateAccountName(newAccountName));
+      emitMetric(METRIC_NAMES.viewPublicKeyAccountRenamed);
+    }
+    setIsEditingName(false);
+  };
+
+  const handleCopy = () => {
+    setIsCopied(true);
+    emitMetric(METRIC_NAMES.viewPublicKeyCopy);
+
+    const t = setTimeout(() => {
+      setIsCopied(false);
+      clearTimeout(t);
+    }, 1000);
+  };
+
+  return (
+    <PopupWrapper>
+      <div className="ViewPublicKey">
+        <BackButton isPopup onClick={() => navigateTo(ROUTES.account)} />
+        <div className="ViewPublicKey__header">
+          {isEditingName ? (
+            <Formik initialValues={initialValues} onSubmit={handleSubmit}>
+              <Form className="ViewPublicKey__form">
+                <Field name="accountName">
+                  {({ field }: FieldProps) => (
+                    <Input
+                      autoComplete="off"
+                      id="accountName"
+                      placeholder={accountName}
+                      {...field}
+                    />
+                  )}
+                </Field>
+                <button
+                  className="ViewPublicKey--account-name-btn"
+                  type="submit"
+                >
+                  <Icon.Check />
+                </button>
+              </Form>
+            </Formik>
+          ) : (
+            <>
+              <div>{accountName}</div>
+              <button
+                className="ViewPublicKey--account-name-btn"
+                onClick={() => setIsEditingName(true)}
+              >
+                <Icon.Edit />
+              </button>
+            </>
+          )}
+        </div>
+
+        <div className="ViewPublicKey__qr-code">
+          <QrCode
+            value={publicKey}
+            style={{
+              width: "10rem",
+              height: "10rem",
+            }}
+          />
+        </div>
+        <div className="ViewPublicKey__address-copy-label">Wallet Address</div>
+        <div className="ViewPublicKey__address-copy">
+          {truncatedPublicKey(publicKey)}
+        </div>
+        <div className="ViewPublicKey__copy-btn">
+          <CopyToClipboard text={publicKey} onCopy={() => handleCopy()}>
```

Comment:
Could you replace this with `CopyText` from SDS?
https://design-system.stellar.org/component/copyTexts

Reply:
did not know I could use a button in there, awesome 😄 
https://github.com/stellar/freighter/pull/375/commits/0055fde0828f27f96d5c255f99c459cd15ba8a0e